resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicDriverFeatureStep.scala

```diff
@@ -51,6 +51,7 @@ private[spark] class BasicDriverFeatureStep(
     .get(DRIVER_MEMORY_OVERHEAD)
     .getOrElse(math.max((conf.get(MEMORY_OVERHEAD_FACTOR) * driverMemoryMiB).toInt,
       MEMORY_OVERHEAD_MIN_MIB))
+  // TODO: Have memory limit checks on driverMemory
```

Comment:
Can we file a JIRA as well and include it in the comment.

Reply:
I wanted to get an opinion from people (@mccheah) into whether or not we wanted to let the K8S api to handle memory limits (via ResourceQuota limit errors) or whether we wanted to catch it in a Spark exception (if we were to include a configuration for memory limits)

Reply:
Hm can you elaborate here? We already set the driver memory limit in this step based on the overhead.

Reply:
Valid point. These are not necessary